theano/tensor/nnet/tests/test_sigm.py

```diff
@@ -237,27 +228,24 @@ def test_local_sigm_times_exp(self):
         def match(func, ops):
             # print [node.op.scalar_op for node in func.maker.fgraph.toposort()]
             assert [node.op for node in func.maker.fgraph.toposort()] == ops
+            assert check_stack_trace(f, ops_to_check=ops)
         m = self.get_mode(excluding=['local_elemwise_fusion', 'inplace'])
         x, y = tensor.vectors('x', 'y')
 
         f = theano.function([x], sigmoid(-x) * tensor.exp(x), mode=m)
-        assert hasattr(f.maker.fgraph.outputs[0].tag, 'trace')
         match(f, [sigmoid])
 
         f = theano.function([x], sigmoid(x) * tensor.exp(-x), mode=m)
-        assert hasattr(f.maker.fgraph.outputs[0].tag, 'trace')
         match(f, [tensor.neg, sigmoid])
```

Comment:
In this case, the `neg` does not have a trace, and it is not obvious to me how to add one given the way the optimization is structured.
For now, I would only to the stack trace check on the sigmoid.